zomes/rea_agent/lib/src/lib.rs

```diff
@@ -27,9 +29,29 @@ pub fn handle_create_agent<S>(entry_def_id: S, agent: CreateRequest) -> RecordAP
     where S: AsRef<str>
 {
     let (header_addr, base_address, entry_resp): (_,_, EntryData) = create_record(&entry_def_id, agent)?;
+    // TODO: check if contains `link_pub_key` field
+    let pub_key = agent_info()?.agent_latest_pubkey;
+    // TODO: error handling to match expect error type
+    let _pub_key_link = create_link(pub_key, header_addr.clone(), HdkLinkType::Any, ())?;
     construct_response(&base_address, header_addr, &entry_resp, /*get_link_fields(&base_address)?*/)
 }
 
+pub fn handle_get_my_agent() -> RecordAPIResult<ResponseData>
+{
+    let my_pub_key = agent_info()?.agent_latest_pubkey;
+    let mut links = get_links(my_pub_key, None)?;
+    let maybe_link = links.pop();
+    // assumes there is only one link
+    match maybe_link {
+        Some(link) => {
+            let header_hash: HeaderHash = link.target.into();
+            let (base_address, entry) = read_record_entry_by_header::<EntryData, EntryStorage, _>(&header_hash)?;
+            construct_response(&base_address, header_hash, &entry, /*get_link_fields(&base_address)?*/)
+        },
+        None => Err(DataIntegrityError::Wasm(WasmError::Guest(String::from("agent pub key link doesn't exist"))))
```

Comment:
we could improve this error message just a bit. It should include info on how to improve the situation, but let's do that once we know the exact strategy for linking

Reply:
@weswalla instead of this,  define a new error variant on `DataIntegrityError`. 

Use something like this one:
```
    #[error("No Agent data is associated with the currently authenticated user")]
    AgentNotLinked,
```

and then use `DataIntegrityError::AgentNotLinked` here instead.

Reply:
Good one. When we eventually have "DNA dependencies" in core there will be referential integrity checks against agent profiles stored in the dependent Agent cell and so those will end up throwing this error as well.